### ISSUE TITLE
ETD supplemental files error

### ISSUE THREAD

**Issue by acozine** [curationexperts/alexandria-v2#344]
When importing /opt/download_root/marc/004511858.xml, which has supplemental files:
```
2015-07-22T20:08:44+00:00  INFO traject (2.1.0) executing with: `-c traject_config.rb /opt/download_root/marc/004511858.xml`
2015-07-22T20:08:44+00:00  INFO Reading from /opt/download_root/marc/004511858.xml
2015-07-22T20:08:44+00:00  INFO    Indexer with 1 processing threads, reader: Traject::MarcReader and writer: ObjectFactoryWriter
  Attaching binary /tmp/d20150722-27889-jsfaoc/MartinezRodriguez_ucsb_0035D_12446.pdf
  Attaching binary /tmp/d20150722-27889-jsfaoc/mussel movie.mp4
  Attaching binary /tmp/d20150722-27889-jsfaoc/mussel on clean glass.mp4
2015-07-22T20:09:14+00:00 FATAL Traject::CommandLine: Unexpected exception, terminating execution: #<Ldp::BadRequest: Error converting InputStream to a BinaryValue>
/opt/alex2/shared/bundle/ruby/2.2.0/gems/ldp-0.3.1/lib/ldp/client/methods.rb:108:in `block in check_for_errors': Error converting InputStream to a BinaryValue (Ldp::BadRequest)
	from /opt/alex2/shared/bundle/ruby/2.2.0/gems/ldp-0.3.1/lib/ldp/client/methods.rb:106:in `tap'
	from /opt/alex2/shared/bundle/ruby/2.2.0/gems/ldp-0.3.1/lib/ldp/client/methods.rb:106:in `check_for_errors'
	from /opt/alex2/shared/bundle/ruby/2.2.0/gems/ldp-0.3.1/lib/ldp/client/methods.rb:89:in `put'
	from /opt/alex2/shared/bundle/ruby/2.2.0/bundler/gems/active_fedora-223391b31c55/lib/active_fedora/caching_connection.rb:24:in `put'
	from /opt/alex2/shared/bundle/ruby/2.2.0/gems/ldp-0.3.1/lib/ldp/resource.rb:72:in `create'
	from /opt/alex2/shared/bundle/ruby/2.2.0/bundler/gems/active_fedora-223391b31c55/lib/active_fedora/file.rb:215:in `create_record'
	from /opt/alex2/shared/bundle/ruby/2.2.0/bundler/gems/active_fedora-223391b31c55/lib/active_fedora/persistence.rb:139:in `create_or_update'
	from /opt/alex2/shared/bundle/ruby/2.2.0/bundler/gems/active_fedora-223391b31c55/lib/active_fedora/persistence.rb:27:in `save'
	from /opt/alex2/shared/bundle/ruby/2.2.0/bundler/gems/active_fedora-223391b31c55/lib/active_fedora/persistence.rb:206:in `block in save_attached_files'
	from /opt/alex2/shared/bundle/ruby/2.2.0/bundler/gems/active_fedora-223391b31c55/lib/active_fedora/persistence.rb:205:in `each'
	from /opt/alex2/shared/bundle/ruby/2.2.0/bundler/gems/active_fedora-223391b31c55/lib/active_fedora/persistence.rb:205:in `save_attached_files'
	from /opt/alex2/shared/bundle/ruby/2.2.0/bundler/gems/active_fedora-223391b31c55/lib/active_fedora/persistence.rb:156:in `update_record'
	from /opt/alex2/shared/bundle/ruby/2.2.0/bundler/gems/active_fedora-223391b31c55/lib/active_fedora/indexing.rb:52:in `update_record'
	from /opt/alex2/shared/bundle/ruby/2.2.0/bundler/gems/active_fedora-223391b31c55/lib/active_fedora/callbacks.rb:243:in `block (2 levels) in update_record'
	from /opt/alex2/shared/bundle/ruby/2.2.0/gems/activesupport-4.2.3/lib/active_support/callbacks.rb:115:in `call'
	from /opt/alex2/shared/bundle/ruby/2.2.0/gems/activesupport-4.2.3/lib/active_support/callbacks.rb:115:in `call'
	from /opt/alex2/shared/bundle/ruby/2.2.0/gems/activesupport-4.2.3/lib/active_support/callbacks.rb:553:in `block (2 levels) in compile'
	from /opt/alex2/shared/bundle/ruby/2.2.0/gems/activesupport-4.2.3/lib/active_support/callbacks.rb:503:in `call'
	from /opt/alex2/shared/bundle/ruby/2.2.0/gems/activesupport-4.2.3/lib/active_support/callbacks.rb:503:in `call'
	from /opt/alex2/shared/bundle/ruby/2.2.0/gems/activesupport-4.2.3/lib/active_support/callbacks.rb:88:in `run_callbacks'
	from /opt/alex2/shared/bundle/ruby/2.2.0/bundler/gems/active_fedora-223391b31c55/lib/active_fedora/callbacks.rb:243:in `block in update_record'
	from /opt/alex2/shared/bundle/ruby/2.2.0/gems/activesupport-4.2.3/lib/active_support/callbacks.rb:115:in `call'
	from /opt/alex2/shared/bundle/ruby/2.2.0/gems/activesupport-4.2.3/lib/active_support/callbacks.rb:115:in `call'
	from /opt/alex2/shared/bundle/ruby/2.2.0/gems/activesupport-4.2.3/lib/active_support/callbacks.rb:553:in `block (2 levels) in compile'
	from /opt/alex2/shared/bundle/ruby/2.2.0/gems/activesupport-4.2.3/lib/active_support/callbacks.rb:503:in `call'
	from /opt/alex2/shared/bundle/ruby/2.2.0/gems/activesupport-4.2.3/lib/active_support/callbacks.rb:503:in `call'
	from /opt/alex2/shared/bundle/ruby/2.2.0/gems/activesupport-4.2.3/lib/active_support/callbacks.rb:88:in `run_callbacks'
	from /opt/alex2/shared/bundle/ruby/2.2.0/bundler/gems/active_fedora-223391b31c55/lib/active_fedora/callbacks.rb:242:in `update_record'
	from /opt/alex2/shared/bundle/ruby/2.2.0/bundler/gems/active_fedora-223391b31c55/lib/active_fedora/persistence.rb:139:in `create_or_update'
	from /opt/alex2/shared/bundle/ruby/2.2.0/bundler/gems/active_fedora-223391b31c55/lib/active_fedora/persistence.rb:27:in `save'
	from /opt/alex2/shared/bundle/ruby/2.2.0/bundler/gems/active_fedora-223391b31c55/lib/active_fedora/validations.rb:50:in `save'
	from /opt/alex2/releases/20150722194038/lib/importer/factory/etd_factory.rb:21:in `after_create'
	from /opt/alex2/releases/20150722194038/lib/importer/factory/object_factory.rb:58:in `block in create'
	from /opt/alex2/shared/bundle/ruby/2.2.0/bundler/gems/active_fedora-223391b31c55/lib/active_fedora/core.rb:36:in `initialize'
	from /opt/alex2/releases/20150722194038/lib/importer/factory/object_factory.rb:56:in `new'
	from /opt/alex2/releases/20150722194038/lib/importer/factory/object_factory.rb:56:in `create'
	from /opt/alex2/releases/20150722194038/lib/importer/factory/object_factory.rb:15:in `run'
	from /opt/alex2/releases/20150722194038/lib/object_factory_writer.rb:70:in `build_object'
	from /opt/alex2/releases/20150722194038/lib/object_factory_writer.rb:64:in `put'
	from /opt/alex2/shared/bundle/ruby/2.2.0/gems/traject-2.1.0/lib/traject/indexer.rb:438:in `block (2 levels) in process'
	from /opt/alex2/shared/bundle/ruby/2.2.0/gems/traject-2.1.0/lib/traject/thread_pool.rb:110:in `block in maybe_in_thread_pool'
	from /opt/alex2/shared/bundle/ruby/2.2.0/gems/concurrent-ruby-0.8.0/lib/concurrent/executor/ruby_thread_pool_worker.rb:62:in `call'
	from /opt/alex2/shared/bundle/ruby/2.2.0/gems/concurrent-ruby-0.8.0/lib/concurrent/executor/ruby_thread_pool_worker.rb:62:in `block in run'
	from /opt/alex2/shared/bundle/ruby/2.2.0/gems/concurrent-ruby-0.8.0/lib/concurrent/executor/ruby_thread_pool_worker.rb:53:in `loop'
	from /opt/alex2/shared/bundle/ruby/2.2.0/gems/concurrent-ruby-0.8.0/lib/concurrent/executor/ruby_thread_pool_worker.rb:53:in `run'
	from /opt/alex2/shared/bundle/ruby/2.2.0/gems/concurrent-ruby-0.8.0/lib/concurrent/executor/ruby_thread_pool_executor.rb:255:in `block in create_worker_thread'
[centos@ip-10-157-142-181 current]$ 

```

**Comment by acozine** [comment]
If you run the importer a second time on the same ETD, it successfully imports all associated files.

**Comment by jcoyne** [comment]
File wasn't rewound. Works fine when `UpdateMetadataFromProquestFile.new(etd).run` is commented out.

**Comment by jcoyne** [comment]
Fixed by 80e318d6af32c378a76b8cbff15437cef95b1845
